javatests/arcs/android/storage/handle/AndroidHandleManagerTest.kt

```diff
@@ -121,107 +126,144 @@ class AndroidHandleManagerTest {
+        assertThat(readBack).containsExactly(entity1, entity2)
+    }
+
+    @Test
+    fun testDereferencingFromSetHandleEntity() = handleManagerTest { hm ->
+        val setHandle = hm.setHandle(setKey, schema)
+        setHandle.store(entity1)
+        setHandle.store(entity2)
+
+        val secondHandle = hm.setHandle(setKey, schema)
+        secondHandle.fetchAll().also { assertThat(it).hasSize(2) }.forEach { entity ->
+            val expectedBestFriend = if (entity.id == "entity1") entity2 else entity1
+            val actualBestFriend = (entity.singletons["best_friend"] as Reference)
+                .dereference()
+            assertThat(actualBestFriend).isEqualTo(expectedBestFriend)
         }
     }
 
     private fun testMapForKey(key: StorageKey) = VersionMap(key.toKeyString() to 1)
 
     @Test
-    fun testSetHandleOnUpdate()  = runBlockingTest {
-        handleManagerTest { hm ->
-            val testCallback1 = mock<SetCallbacks<RawEntity>>()
-            val testCallback2 = mock<SetCallbacks<RawEntity>>()
-            val firstHandle = hm.setHandle(setKey, schema, testCallback1)
-            val secondHandle = hm.setHandle(setKey, schema, testCallback2)
-
-            val expectedAdd = CrdtSet.Operation.Add(
-                setKey.toKeyString(),
-                testMapForKey(setKey),
-                entity1
-            )
-            secondHandle.store(entity1)
-            verify(testCallback1, times(1)).onUpdate(firstHandle, expectedAdd)
-            verify(testCallback2, times(1)).onUpdate(secondHandle, expectedAdd)
-
-            firstHandle.remove(entity1)
-            val expectedRemove = CrdtSet.Operation.Remove(
-                setKey.toKeyString(),
-                testMapForKey(setKey),
-                entity1
-            )
-            verify(testCallback1, times(1)).onUpdate(firstHandle, expectedRemove)
-            verify(testCallback2, times(1)).onUpdate(secondHandle, expectedRemove)
-        }
+    fun testSetHandleOnUpdate() = handleManagerTest { hm ->
+        val testCallback1 = mock<SetCallbacks<RawEntity>>()
+        val testCallback2 = mock<SetCallbacks<RawEntity>>()
+        val firstHandle = hm.setHandle(setKey, schema, testCallback1)
+        val secondHandle = hm.setHandle(setKey, schema, testCallback2)
+
+        val expectedAdd = CrdtSet.Operation.Add(
+            setKey.toKeyString(),
+            testMapForKey(setKey),
+            entity1
+        )
+        secondHandle.store(entity1)
+        verify(testCallback1, times(1)).onUpdate(firstHandle, expectedAdd)
+        verify(testCallback2, times(1)).onUpdate(secondHandle, expectedAdd)
+
+        firstHandle.remove(entity1)
+        val expectedRemove = CrdtSet.Operation.Remove(
+            setKey.toKeyString(),
+            testMapForKey(setKey),
+            entity1
+        )
+        verify(testCallback1, times(1)).onUpdate(firstHandle, expectedRemove)
+        verify(testCallback2, times(1)).onUpdate(secondHandle, expectedRemove)
```

Comment:
No need for `times(1)`, that's the default (here and throughout)

Reply:
FWIW, these were pre-existing (as was the `hm`).. I'll leave it for now

Reply:
For the curious:

`hm` is because I spent the last 3 years doing go development. :-) I am fixing that in a cleanup PR I have above this.

The `times(1)` was intentional... it was useful for me as a newcomer to the `verify` API to be a bit more explicit. I can remove it in a subsequent cleanup.

Reply:
Thanks! I thought it looked suspiciously like Go :stuck_out_tongue: 

In Java we recommend omitting the `times(1)` -- it's unnecessary extra code, and it can mislead readers of the code into thinking that the default is *not* 1.